isthmus/src/test/java/io/substrait/isthmus/SubstraitRelNodeConverterTest.java

```diff
@@ -0,0 +1,242 @@
+package io.substrait.isthmus;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+
+import io.substrait.dsl.SubstraitBuilder;
+import io.substrait.plan.Plan;
+import io.substrait.relation.Rel;
+import io.substrait.relation.Set.SetOp;
+import io.substrait.type.Type;
+import io.substrait.type.TypeCreator;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.calcite.rel.type.RelDataType;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+public class SubstraitRelNodeConverterTest extends PlanTestBase {
+
+  static final TypeCreator R = TypeCreator.of(false);
+  static final TypeCreator N = TypeCreator.of(true);
+
+  final SubstraitBuilder b = new SubstraitBuilder(extensions);
+
+  // Define a shared table (i.e. a NamedScan) for use in tests.
+  final List<Type> commonTableType = List.of(R.I32, R.FP32, N.STRING, N.BOOLEAN);
+  final List<Type> commonTableTypeTwice =
+      Stream.concat(commonTableType.stream(), commonTableType.stream())
+          .collect(Collectors.toList());
+  final Rel commonTable =
+      b.namedScan(List.of("example"), List.of("a", "b", "c", "d"), commonTableType);
+
+  final SubstraitToCalcite converter = new SubstraitToCalcite(extensions, typeFactory);
+
+  void assertRowMatch(RelDataType actual, Type... expected) {
+    assertRowMatch(actual, Arrays.asList(expected));
+  }
+
+  void assertRowMatch(RelDataType actual, List<Type> expected) {
+    Type type = TypeConverter.convert(actual);
+    assertInstanceOf(Type.Struct.class, type);
+    Type.Struct struct = (Type.Struct) type;
+    assertEquals(expected, struct.fields());
+  }
+
+  @Nested
+  class Aggregate {
```

Comment:
Using Nested classes here with the idea that more Rel specific tests can/will be added in the future.